src/web/handlers.rs

```diff
@@ -431,3 +431,31 @@ pub fn heartbeat(hb: HeartbeatRequest) -> impl Future<Item = HttpResponse, Error
         }
     })
 }
+
+pub fn test_error(
+    _req: HttpRequest,
+    ter: TestErrorRequest,
+) -> impl Future<Item = HttpResponse, Error = ApiError> {
+    // generate an error for sentry.
+    use crate::error::ApiErrorKind;
+
+    /*  The various error log macros only can take a string.
+        Additional values can be expressed as KV (key value) after a `;`
+        e.g.
+        ```
+        error!("Something Bad {:?}", err;
+                "ua.os.family" => wtags.get("ua.os.family"),
+                "ua.browser.family" => wtags.get("ua.browser.family"),
+                "ua.name" => wtags.get("ua.name"),
+                "ua.os.ver" => wtags.get("ua.os.ver"),
+                "ua.browser.ver" => wtags.get("ua.browser.ver"))
+        ```
+        TODO: find some way to transform Tags into error::KV
```

Comment:
KV's a trait that Tags can implement. [slog-derive](https://github.com/slog-rs/derive) makes it easy.

Probably better to tackle that in a separate issue.

Reply:
see #387